src/EventStore.Core/Services/Gossip/GossipServiceBase.cs

```diff
@@ -383,5 +401,16 @@ public abstract class GossipServiceBase : IHandle<SystemMessage.SystemInit>,
 				, newMembers
 			);
 		}
+		
+		private void PublishIfClusterHasMultipleVersions(ClusterInfo cluster) {
+			List<MemberInfo> aliveMembers = cluster.Members.Where(memberInfo => memberInfo.IsAlive).ToList();
+			int numDistinctVersions = aliveMembers.Select(memberInfo => memberInfo.ESVersion).Where(esVersion => !VersionInfo.UnknownVersion.Equals(esVersion)).Distinct().Count();
+
+			if (numDistinctVersions > 1) {
+				Dictionary<EndPoint, string> ipAndVersion = aliveMembers.ToDictionary(memberInfo => memberInfo.HttpEndPoint,
+					memberInfo => memberInfo.ESVersion);
+				_bus.Publish(new ClusterStateMessage.MultipleVersionsOnNodes(ipAndVersion));
```

Comment:
It looks like this is done for testing, would it be an idea to have a separate class for this logic and use/subscribe to already available messages?